### PR TITLE
sysvar: add value limitation for net_buffer_length

### LINKED ISSUE #31421: max_allowed_packet and net_buffer_length   session | global scope difference in tidb
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
set  max_allowed_packet=100;
set  net_buffer_length=100;
set global max_allowed_packet=100;
set global net_buffer_length=100;
select @@net_buffer_length, @@max_allowed_packet;
```
### 2. What did you expect to see? (Required)
```sql
mysql> set  max_allowed_packet=100;
ERROR 1621 (HY000): SESSION variable 'max_allowed_packet' is read-only. Use SET GLOBAL to assign the value
mysql> set  net_buffer_length=100;
ERROR 1621 (HY000): SESSION variable 'net_buffer_length' is read-only. Use SET GLOBAL to assign the value
mysql> set global max_allowed_packet=100;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> set global net_buffer_length=100;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> select @@net_buffer_length, @@max_allowed_packet;
+---------------------+----------------------+
| @@net_buffer_length | @@max_allowed_packet |
+---------------------+----------------------+
|                1024 |                 1024 |
+---------------------+----------------------+
1 row in set (0.00 sec)



```
### 3. What did you see instead (Required)
```sql
mysql> set  max_allowed_packet=100;
Query OK, 0 rows affected, 1 warning (0.00 sec)
mysql> set  net_buffer_length=100;
Query OK, 0 rows affected (0.00 sec)
mysql> set global max_allowed_packet=100;
Query OK, 0 rows affected, 1 warning (0.01 sec)

mysql> set global net_buffer_length=100;
Query OK, 0 rows affected (0.00 sec)

mysql> select @@net_buffer_length, @@max_allowed_packet;
+---------------------+----------------------+
| @@net_buffer_length | @@max_allowed_packet |
+---------------------+----------------------+
| 100                 |                 1024 |
+---------------------+----------------------+
1 row in set (0.00 sec)


```
### 4. What is your TiDB version? (Required)

```sql
| Release Version: v5.5.0-alpha-21-g722303bca-dirty
Edition: Community
Git Commit Hash: 722303bca8d279ff7c9d87e942095d320499f47e
Git Branch: master
UTC Build Time: 2022-01-05 19:11:01
GoVersion: go1.16.9
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
```

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #31421 

Problem Summary:

Sysvar `net_buffer_length` has a minimum and maximum in MySQL, but TiDB lacks this limitation.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by morgo — APPROVED** [review]
LGTM

From the bug report, MySQL also refuses to allow netbufferlength/maxallowedpacket on a session basis. You could try adding a Validation func which returns an error if the scope is session.

But I am okay with just this change as well.

File: sessionctx/variable/sysvar_test.go [hunk 1]
```diff
@@ -843,3 +843,19 @@ func TestNoValidateForNoop(t *testing.T) {
...
 }
+
+func TestNetBufferLength(t *testing.T) {
+	netBufferLength := GetSysVar(NetBufferLength)
+	vars := NewSessionVars()
+	vars.GlobalVarsAccessor = NewMockGlobalAccessor4Tests()
+
+	val, err := netBufferLength.Validate(vars, "1", ScopeGlobal)
+	require.NoError(t, err)
+	require.Equal(t, val, "1024") // converts it to min value
```
Comment (hawkingrei) [inline sessionctx/variable/sysvar_test.go:854]: You need to change the order of parameter in the ```require.Equal```.
```
require.Equal(t, expected, actual)
```

**Review by hawkingrei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**